pkg/es/config/config.go

```diff
@@ -44,37 +44,59 @@ import (
-	PrioritySpanTemplate           int64          `mapstructure:"priority_span_template"`
-	PriorityServiceTemplate        int64          `mapstructure:"priority_service_template"`
-	PriorityDependenciesTemplate   int64          `mapstructure:"priority_dependencies_template"`
-	Timeout                        time.Duration  `mapstructure:"-"`
-	BulkSize                       int            `mapstructure:"-"`
-	BulkWorkers                    int            `mapstructure:"-"`
-	BulkActions                    int            `mapstructure:"-"`
-	BulkFlushInterval              time.Duration  `mapstructure:"-"`
-	IndexPrefix                    string         `mapstructure:"index_prefix"`
-	IndexDateLayoutSpans           string         `mapstructure:"-"`
-	IndexDateLayoutServices        string         `mapstructure:"-"`
-	IndexDateLayoutSampling        string         `mapstructure:"-"`
-	IndexDateLayoutDependencies    string         `mapstructure:"-"`
-	IndexRolloverFrequencySpans    string         `mapstructure:"-"`
-	IndexRolloverFrequencyServices string         `mapstructure:"-"`
+	Servers                      []string      `mapstructure:"server_urls" valid:"required,url"`
+	RemoteReadClusters           []string      `mapstructure:"remote_read_clusters"`
+	Username                     string        `mapstructure:"username"`
+	Password                     string        `mapstructure:"password" json:"-"`
+	TokenFilePath                string        `mapstructure:"token_file"`
+	PasswordFilePath             string        `mapstructure:"password_file"`
+	AllowTokenFromContext        bool          `mapstructure:"-"`
+	Sniffer                      bool          `mapstructure:"sniffer"` // https://github.com/olivere/elastic/wiki/Sniffing
+	SnifferTLSEnabled            bool          `mapstructure:"sniffer_tls_enabled"`
+	MaxDocCount                  int           `mapstructure:"-"` // Defines maximum number of results to fetch from storage per query
+	MaxSpanAge                   time.Duration `mapstructure:"-"` // configures the maximum lookback on span reads
+	NumShards                    int64         `mapstructure:"num_shards"`
+	NumReplicas                  int64         `mapstructure:"num_replicas"`
+	PrioritySpanTemplate         int64         `mapstructure:"priority_span_template"`
+	PriorityServiceTemplate      int64         `mapstructure:"priority_service_template"`
+	PriorityDependenciesTemplate int64         `mapstructure:"priority_dependencies_template"`
+	Timeout                      time.Duration `mapstructure:"-"`
+	BulkSize                     int           `mapstructure:"-"`
+	BulkWorkers                  int           `mapstructure:"-"`
+	BulkActions                  int           `mapstructure:"-"`
+	BulkFlushInterval            time.Duration `mapstructure:"-"`
+	IndexPrefix                  string        `mapstructure:"index_prefix"`
+	// Deprecated: use Indices and IndexOptions instead.
+	IndexDateLayoutSpans string `mapstructure:"-"`
```

Comment:
there is no need to keep deprecated fields in a struct, we can just change the code not to refer to them. And you need to change all the code that previously referred to the old fields to refer to the new ones.

Reply:
got it.